react/ProductDetails.js

```diff
@@ -179,249 +135,224 @@ class ProductDetails extends Component {
+    styles: productPriceLoaderStyles,
+    listPriceContainerClass: 't-small-s t-small-ns c-muted-2 mb2',
+    sellingPriceLabelClass: 't-heading-6-s t-heading-5-ns dib',
+    listPriceLabelClass: 'dib strike',
+    listPriceClass: 'ph2 dib strike',
+    sellingPriceContainerClass: 'pv1 b c-on-base',
+    sellingPriceClass: 't-heading-2-s dib ph2',
+    installmentContainerClass: 't-mini-s t-small-ns c-on-base',
+    installmentClass: 't-body',
+    interestRateClass: 'dib ph2',
+    savingsContainerClass: 'c-success mt3',
+    savingsClass: 'dib t-small',
+    loaderClass: 'h4-s mw6-s pt2-s',
+    listPrice: path(['ListPrice'], commertialOffer()),
+    sellingPrice: path(['Price'], commertialOffer()),
+    installments: path(['Installments'], commertialOffer()),
+    ...price,
+  }
 
-    const availableQuantity = path(['AvailableQuantity'], this.commertialOffer)
-    const showProductPrice =
-      Number.isNaN(+availableQuantity) || availableQuantity > 0
+  const availableQuantity = path(['AvailableQuantity'], commertialOffer())
+  const showProductPrice =
+    Number.isNaN(+availableQuantity) || availableQuantity > 0
 
-    return (
-      <Container
-        className={`${productDetails.container} pt6 pb8 justify-center flex`}
-      >
-        <div className="w-100 mw9">
-          <article className="mb7">
-            {categories && (
-              <ExtensionPoint
-                id="breadcrumb"
-                term={path(['productName'], product)}
-                categories={categories}
-              />
-            )}
+  return (
+    <Container
+      className={`${productDetails.container} pt6 pb8 justify-center flex`}
+    >
+      <div className="w-100 mw9">
+        <article className="mb7">
+          {categories && (
+            <ExtensionPoint
+              id="breadcrumb"
+              term={path(['productName'], product)}
+              categories={categories}
+            />
+          )}
 
-            <div
-              className={`${productDetails.nameContainer} c-on-base mb4 dn-l`}
-            >
-              <ExtensionPoint id="product-name" {...productNameProps} />
-            </div>
-            <div className="flex flex-wrap">
-              <div className="w-60-l w-100">
-                <div className="fr-ns w-100 h-100">
-                  <div className="flex justify-center">
-                    <ExtensionPoint
-                      id="product-images"
-                      images={this.getImages()}
-                    />
-                  </div>
+          <div className={`${productDetails.nameContainer} c-on-base mb4 dn-l`}>
+            <ExtensionPoint id="product-name" {...productNameProps} />
+          </div>
+          <div className="flex flex-wrap">
+            <div className="w-60-l w-100">
+              <div className="fr-ns w-100 h-100">
+                <div className="flex justify-center">
+                  <ExtensionPoint id="product-images" images={getImages()} />
```

Comment:
Do not call functions like that on component props. Could cause undesired effects.